Source/Scene/Primitive.js

```diff
@@ -1217,6 +1218,10 @@ define([
             modelMatrix = this.modelMatrix;
         }
 
+        if (!Matrix4.equals(this.modelMatrix, Matrix4.IDENTITY) && ((this._numberOfInstances > 1 && !this._validModelMatrix) || (frameState.mode === SceneMode.SCENE2D || frameState.mode === SceneMode.COLUMBUS_VIEW))) {
```

Comment:
I would change `(frameState.mode === SceneMode.SCENE2D || frameState.mode === SceneMode.COLUMBUS_VIEW)` to `(frameState.mode !== SceneMode.SCENE3D)`, this will cover morphing and is more specific.

Also, wrap the if block in a pragma statement, like we do with other developer errors.  See [here](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Scene/Primitive.js#L759) for an example.